### ISSUE TITLE
rotating_file_handler unable to log to /dev/null

### ISSUE THREAD

**Issue by dirocco** [odygrd/quill#297]
After upgrading from 1.6.3 to 2.9.0, I'm no longer able to use "/dev/null" as a file name for `quill::rotating_file_hander`. This is useful for exercising quill and custom fmt types during tests without actually writing to disk.

The error is in `quill/src/detail/misc/FileUtilities.cpp` where `std::filesystem::file_size` throws an uncaught exception when `RotatingFileHandler.cpp` tries to set `_current_size`:
`size_t file_size(fs::path const& filename) { return static_cast<size_t>(fs::file_size(filename)); }`

Here is the exception text:
`filesystem error: cannot get file size: Operation not supported [/dev/null]`

**Comment by odygrd** [comment]
hey, thanks for reporting this. I have added support to RotatingFileHandler and TimeRotatingFileHandler for /dev/null in latest master.

I have also added the `NullHandler` class.

Another way to skip logs completely is to never call `quill::start()` in your testing, that way the logs are only pushed to the unbounded queues but are never processed. But for more complete regression testing it's better to use /dev/null or the NullHandler as it will also process the log statements during your testing.
